src/Corax/Queries/MoreLikeThisQuery.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Corax.Pipeline;
+using Sparrow.Json;
+
+namespace Corax.Queries
+{
+    public class MoreLikeThisQuery 
+    {
+        public static IQueryMatch Build(IndexSearcher searcher, BlittableJsonReaderObject doc, Analyzer analyzer)
+        {
+            IQueryMatch match = TermMatch.CreateEmpty();
+            BlittableJsonReaderObject.PropertyDetails prop = default;
+            for (int i = 0; i < doc.Count; i++)
+            {
+                doc.GetPropertyByIndex(i, ref prop);
+                switch (prop.Token & BlittableJsonReaderBase.TypesMask)
+                {
+                    case BlittableJsonToken.Integer:
+                    case BlittableJsonToken.LazyNumber:
+                    case BlittableJsonToken.Boolean:
+                        match = searcher.Or(match, searcher.TermQuery(prop.Name, prop.Value.ToString()));
+                        break;
+                    case BlittableJsonToken.String:
+                        match = searcher.Or(match, CreateAnalyzedQuery(prop.Name, searcher, (LazyStringValue)prop.Value, analyzer));
+                        break;
+                    case BlittableJsonToken.CompressedString:
+                        match = searcher.Or(match, CreateAnalyzedQuery(prop.Name, searcher, ((LazyCompressedStringValue)prop.Value).ToLazyStringValue(), analyzer));
+                        break;
+                }
+            }
+
+            return searcher.OrderByScore( searcher.Boost(match, 2));
```

Comment:
@redknightlois this doesn't actually does what I want it to (boost by relevance), however, it looks like I'm not using the API properly

Reply:
You are boosting the entire match there. Which means that you are scoring the output of the whole query in there, not the specific part of it.

Reply:
What you want to do is decorate the `TermQuery` with the term frequency score function. 

``` 
searcher.Boost(searcher.TermQuery(prop.Name, prop.Value.ToString()), default(TermFrequencyScoreFunction));
```